### PR TITLE
Input Component 

### PR DESCRIPTION
**Pull request by yosefalnajjarofficial**
- #### Added the component 
   - Input: takes props for the type, name, placeholder, etc and it's a controlled component
- #### Added some styles to it 
   - Gave all the inputs a default style by giving them the .custom-input class
#42 

### To use the input:
- Import the component in the app 
```js
import Input from './components/commonComponents/Input';
```
- Add props to the component 
```js
 <Input
          type="text"
          name="username"
          className="username"
          label="username"
          placeholder="Enter your username"
          value=""
          onChange=""
          errMsg=""
        />
```

### CODE REVIEW

File: client/src/App.css [hunk 1]
```diff
@@ -0,0 +1,7 @@
+body {
+  padding: 0;
+  margin: 0;
+  -webkit-box-sizing: border-box;
+          box-sizing: border-box;
+}
+/*# sourceMappingURL=App.css.map */
```
Comment (Fatmasiam) [inline client/src/App.css]: remove useless lines

File: client/src/components/commonComponents/Input/index.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  margin: 10px 0;
+  padding: 10px 20px;
+  background-color: #DADADA;
+  border: none;
+  outline: none;
+  font-size: 18px;
+  color: #8A8A8A;
+  text-transform: capitalize;
+}
+/*# sourceMappingURL=index.css.map */
```
Comment (Fatmasiam) [inline client/src/components/commonComponents/Input/index.css]: remove useless lines

File: client/src/components/commonComponents/Input/index.css [hunk 3]
```diff
@@ -0,0 +1,12 @@
+input {
```
Comment (Fatmasiam) [inline client/src/components/commonComponents/Input/index.css]: don't use the tag name in the CSS

File: .gitignore [hunk 4]
```diff
@@ -1,7 +1,10 @@
 node_modules
+<<<<<<< HEAD
+=======
 config.env
 node_modules
 .nyc_output
 .vscode
 .env
 coverage
+>>>>>>> d5ae29f2e0b6364199d405467c737eb3acbc8e33
```
Comment (Fatmasiam) [inline .gitignore]: conflict !!!

File: client/src/components/commonComponents/Input/index.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+const Input = ({ type, name, label, placeholder, value, onChange }) => {
+  return (
+    <input
+      type={type}
+      name={name}
+      aria-label={label}
+      placeholder={placeholder}
+      value={value}
+      onChange={onChange}
```
Comment (Fatmasiam) [inline client/src/components/commonComponents/Input/index.js]: how can I make the style to th input --> add className attr

Comment (Fatmasiam) [reply client/src/components/commonComponents/Input/index.js]: and add span for the error msg

File: client/src/components/commonComponents/Input/index.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './index.css';
+
+const Input = ({ type, name, label, placeholder, value, onChange }) => {
+  return (
+    <input
+      type={type}
+      name={name}
+      aria-label={label}
```
Comment (Fatmasiam) [inline client/src/components/commonComponents/Input/index.js]: remove this and try to wrap the input with Lable

**Comment by Fatmasiam** [comment]
try to solve the conflict then the Travis will be solved

File: client/src/App.css (hunk 1 above)
Comment (yosefalnajjarofficial) [reply client/src/App.css]: I removed it you can check it now

File: client/src/components/commonComponents/Input/index.css (hunk 2 above)
Comment (yosefalnajjarofficial) [reply client/src/components/commonComponents/Input/index.css]: I did it

File: client/src/components/commonComponents/Input/index.css (hunk 3 above)
Comment (yosefalnajjarofficial) [reply client/src/components/commonComponents/Input/index.css]: I removed the names of the tags and put classes instead

File: .gitignore (hunk 4 above)
Comment (yosefalnajjarofficial) [reply .gitignore]: I solved the conflict

File: client/src/components/commonComponents/Input/index.js (hunk 5 above)
Comment (yosefalnajjarofficial) [reply client/src/components/commonComponents/Input/index.js]: I added both

File: client/src/components/commonComponents/Input/index.js (hunk 6 above)
Comment (yosefalnajjarofficial) [reply client/src/components/commonComponents/Input/index.js]: I wrapped the input within a div

**Pull request merged**